### PR TITLE
[TASK] Use non composer mode for extensions

### PR DESCRIPTION
**Pull request by gilbertsoft**
With the change https://review.typo3.org/c/Packages/TYPO3.CMS/+/72071
functional and acceptance tests are not longer working. Therefor the
mode is changed to non composer for these tests.

### CODE REVIEW

File: Classes/Core/SystemEnvironmentBuilder.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
```
Comment (oliverklee) [inline Classes/Core/SystemEnvironmentBuilder.php:3]: I propose we make new files strict to begin with.

Comment (gilbertsoft) [reply Classes/Core/SystemEnvironmentBuilder.php:5]: Oops

File: Classes/Core/Testbase.php [hunk 2]
```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
...
+            }
+            if (!symlink($realTarget, $link)) {
+                throw new \RuntimeException(
+                    sprintf('Creating link failed (%s -> %s).', $link, $target),
                     1376657199
                 );
             }
         }
+
+        self::setUpEntryPoints($instancePath);
```
Comment (helhum) [inline Classes/Core/Testbase.php:214]: must be $this->

File: Classes/Core/Testbase.php [hunk 3]
```diff
@@ -529,7 +623,12 @@ public function setUpPackageStates(
      */
     private function isPackageArtifactNeeded(): bool
     {
-        return defined('TYPO3_COMPOSER_MODE') && class_exists(PackageCacheEntry::class);
+        return false;
+
+        // For the time being we do override composer mode and this code is not
+        // needed anymore but kept here for later usage once real composer tests
+        // get included.
+        //return defined('TYPO3_COMPOSER_MODE') && class_exists(PackageCacheEntry::class);
```
Comment (helhum) [inline Classes/Core/Testbase.php:631]: We have version control. If we don't need code any more, we can safely remove it 😅

File: Classes/Core/Testbase.php [hunk 4]
```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
...
+
+        $this->defineOriginalRootPath();
+
+        /** @var array<string, string> $linksToSet */
         $linksToSet = [
-            '../../../../' => $instancePath . '/typo3_src',
-            'typo3_src/typo3' => $instancePath . '/typo3',
-            'typo3_src/index.php' => $instancePath . '/index.php',
+            ORIGINAL_ROOT => $instancePath . '/typo3_src',
+            'typo3_src/typo3/sysext/' => $instancePath . '/typo3/sysext',
```
Comment (helhum) [inline Classes/Core/Testbase.php:192]: Nothing we can tackle with this change, but this code will have a hard time when typo3 extensions are installed in vendor ...

File: Classes/Core/Testbase.php [hunk 5]
```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
...
-     * @return void
+     * @throws \InvalidArgumentException
+     * @throws \RuntimeException
      */
-    public function setUpInstanceCoreLinks($instancePath): void
+    public function setUpInstanceCoreLinks(string $instancePath): void
     {
+        $filesystem = new Filesystem();
+
+        if (!$filesystem->isAbsolutePath($instancePath)) {
```
Comment (helhum) [inline Classes/Core/Testbase.php:180]: is there a reason you introduce this check here now?

File: Classes/Core/Testbase.php [hunk 6]
```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
...
+     *
+     * @param string $instancePath Absolute path to test instance
+     * @throws \InvalidArgumentException
+     * @throws \UnexpectedValueException
+     */
+    public function setUpEntryPoints(string $instancePath): void
+    {
+        $filesystem = new Filesystem();
+
+        if (!$filesystem->isAbsolutePath($instancePath)) {
```
Comment (helhum) [inline Classes/Core/Testbase.php:230]: I don't think we need to double check this though

File: Classes/Core/Testbase.php [hunk 7]
```diff
@@ -164,29 +166,121 @@ public function removeOldInstanceIfExists($instancePath): void
...
+        }
+
+        $entryPointsToSet = [
+            InstalledVersions::getInstallPath('typo3/cms-backend') . '/Resources/Private/Php/backend.php' => $instancePath . '/typo3/index.php',
+            InstalledVersions::getInstallPath('typo3/cms-frontend') . '/Resources/Private/Php/frontend.php' => $instancePath . '/index.php',
+            InstalledVersions::getInstallPath('typo3/cms-install') . '/Resources/Private/Php/install.php' => $instancePath . '/typo3/install.php',
+        ];
+
+        ;
+        if (($installersPath = InstalledVersions::getInstallPath('typo3/cms-composer-installers')) === null) {
```
Comment (helhum) [inline Classes/Core/Testbase.php:244]: we should either check for all path whether they are null or none of them imho

File: Classes/Core/Testbase.php (hunk 5 above)
Comment (gilbertsoft) [reply Classes/Core/Testbase.php:180]: Yes, if it's not absolute we can get in trouble with symlinks created in the wrong folders.

**Comment by gilbertsoft** [comment]
@helhum Found two other places where I have replaced the SysEnvBuilder.

**Review by bmack — APPROVED** [review]

**Comment by helhum** [comment]
Thanks for the merge!

**Pull request merged**